### PR TITLE
add representation method for notebooks for BackendStatus

### PR DESCRIPTION
**Pull request by peendebak**
### Summary

This PR adds a representation method for the `BackendStatus` object. The representation is implemented using `_repr_html_` and is used automatically in Jupyter notebook.

### Details and comments

An image showing the old and new representation:

image

There a several other objects that might benefit from a representation method, these are not addressed in this PR.

### CODE REVIEW

File: qiskit/providers/models/backendstatus.py [hunk 1]
```diff
@@ -67,3 +68,14 @@ def __eq__(self, other):
...
         return False
+
+    def _repr_html_(self):
+        rpr = self.__repr__()
+        txt = ''
+        txt += f'<pre>{html.escape(rpr)}</pre>'
+        txt += f'<b>name</b>: {self.backend_name}<br/>'
+        txt += f'<b>version</b>: {self.backend_version},'
+        txt += f' <b>pending jobs</b>: {self.pending_jobs}<br/>'
+        txt += f'<b>status</b>: {self.status_msg}<br/>'
```
Comment (taalexander) [inline qiskit/providers/models/backendstatus.py:79]: Prefer a multiline f-string - https://realpython.com/python-f-strings/#multiline-f-strings.

**Review by taalexander — COMMENTED** [review]
Thoughts on this @nonhermitian? What is the appropriate testing strategy for this functionality?

**Comment by nonhermitian** [comment]
So I think it is fine with respect to what one wants to accomplish.  However, it is not really a general piece of code; not all Qiskit backends support status or the specific status fields like `pending_jobs`.  This would work for IBM systems but not , for example, IONQ (as it stands right now).  So I am wondering if Qiskit should not just say the backend will have a status, but then defer to the providers as to what this status actually is?  Perhaps being operational or not is the bare minimum.

File: qiskit/providers/models/backendstatus.py (hunk 1 above)
Comment (eendebakpt) [reply qiskit/providers/models/backendstatus.py]: Done

**Comment by eendebakpt** [comment]
@nonhermitian For the representation I used the fields that are available in the `BackendStatus` object. I would assume each backend sets these values (or sets to a dummy such as `''`)

**Comment by taalexander** [comment]
@nonhermitian are you ok with this moving forward?

**Comment by nonhermitian** [comment]
I think everything here is technically part of the spec, so it should be fine.

File: qiskit/providers/models/backendstatus.py [hunk 2]
```diff
@@ -66,3 +74,16 @@ def __eq__(self, other):
             if self.__dict__ == other.__dict__:
                 return True
         return False
+
+    def _repr_html_(self) -> str:
+        """Return html representation of the object"""
```
Comment (mtreinish) [inline qiskit/providers/models/backendstatus.py:79]: Just a small nit, I would probably put something here about this only being used in ipython and/or jupyter. But not worth blocking over if everything else looks good.

File: qiskit/providers/models/backendstatus.py [hunk 3]
```diff
@@ -66,3 +74,16 @@ def __eq__(self, other):
...
+        rpr = self.__repr__()
+        html_code = (
+            f"<pre>{html.escape(rpr)}</pre>"
+            f"<b>name</b>: {self.backend_name}<br/>"
+            f"<b>version</b>: {self.backend_version},"
+            f" <b>pending jobs</b>: {self.pending_jobs}<br/>"
```
Comment (taalexander) [inline qiskit/providers/models/backendstatus.py:85]: Was it intentional to display these on the same line?

**Review by taalexander — COMMENTED** [review]
It is hard to test code of this form without significant investment in mocking. While we would appreciate tests of this form after discussion with @mtreinish we would ok moving forward without unittests as I have validated that it works for me locally.

File: qiskit/providers/models/backendstatus.py (hunk 2 above)
Comment (peendebak) [reply qiskit/providers/models/backendstatus.py]: Done

**Comment by peendebak** [comment]
> It is hard to test code of this form without significant investment in mocking. While we would appreciate tests of this form after discussion with @mtreinish we would ok moving forward without unittests as I have validated that it works for me locally.

The `BackendStatus` object itself was not in the unit tests (except for checking the schema). I added a very simple test (construction of the `BackendStatus`, executing of the repr methods)

File: qiskit/providers/models/backendstatus.py (hunk 3 above)
Comment (peendebak) [reply qiskit/providers/models/backendstatus.py:90]: Yes, to safe some space in the representation. But I can make two lines out of it if that seems better.

**Comment by eendebakpt** [comment]
@taalexander Addressed the comments

**Review by taalexander — APPROVED** [review]
Thanks @peendebak this is excellent 😄

**Pull request merged**